gokart/task.py

```diff
@@ -60,6 +60,13 @@ def output(self):
         file_path = self.__module__.replace(".", "/")
         return self.make_target(os.path.join(file_path, f"{type(self).__name__}.pkl"))
 
+    def requires(self):
+        tasks = self.make_task_instance_dictionary()
+        return tasks or []  # when tasks is empty dict, then this returns empty list.
+
+    def make_task_instance_dictionary(self) -> Dict[str, 'TaskOnKart']:
+        return {key[:-5]: var for key, var in vars(self).items() if key.endswith('_task') and isinstance(var, TaskOnKart)}
```

Comment:
[ASK] What is the intention requiring "_task" for identifier? I think just checking `TaskOnKart ` type can do.

Reply:
I prefere
```python
SomeTask(some_data_task=SomeDataTask())
...
some_data = self.load_data_frame('some_data_task')
```
than
```python
SomeTask(some_data=SomeDataTask())
...
some_data = self.load_data_frame('some_data')
```

Reply:
I think ignoring task without specific prefix is a little bit confusing.
How about following alternative?

```suggestion
        return {re.sub(r'_model$', '', key) var for key, var in vars(self).items() if isinstance(var, TaskOnKart)}
```